### PR TITLE
Microservices Testing with WireMock

### PR DESCRIPTION
**Pull request by otrosien**

### CODE REVIEW

File: _posts/2016/2016-07-07-wiremock.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+---
+layout: post
+title: "How not to produce 'Distributed Balls of Mud'"
+date: "2016-07-07 12:00:00"
+image: /blog-header/balls-of-mud.jpg
```
Comment (BirgitBader) [inline _posts/2016/2016-07-07-wiremock.md]: I adapted the image size to our needs. Please use the one attached:
balls-of-mud

File: _posts/2016/2016-07-07-wiremock.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+categories: tech-stories
+authors: ["Oliver"]
```
Comment (BirgitBader) [inline _posts/2016/2016-07-07-wiremock.md]: Oliver T.

File: _posts/2016/2016-07-07-wiremock.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
+---
+layout: post
+title: "How not to produce 'Distributed Balls of Mud'"
+date: "2016-07-07 12:00:00"
```
Comment (BirgitBader) [inline _posts/2016/2016-07-07-wiremock.md]: 2016-07-14

Please also change the .md file respectively.

File: _posts/2016/2016-07-07-wiremock.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
+---
```
Comment (BirgitBader) [inline _posts/2016/2016-07-07-wiremock.md]: Please branch from 'master' and PR onto 'master' (only valid for blog posts).

File: _posts/2016/2016-07-07-wiremock.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+image: /blog-header/balls-of-mud.jpg
+categories: tech-stories
+authors: ["Oliver"]
+---
+
+There is a lot of buzz about microservices these days, and here is our take on managing decoupled microservices, while still keeping confident in REST API compatibility.
+
+One of the SOA missteps was trying to enforce a canonical domain model, which lead to bloated domain objects, that still would not fit all use cases and needed some sort of central management.
+Just try to come up with a common model of any real-world object and you'll know that this does not scale the more people are involved.
+In a more abstract sense, it is our strive to perfection and the strive of perfect modeling of our problem domain that drove us into the wrong direction.
```
Comment (BirgitBader) [inline _posts/2016/2016-07-07-wiremock.md]: modeling > modelling 

(British English)

File: _posts/2016/2016-07-07-wiremock.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+David Dawson phrased this as "Perfection and How It Ruined The World" in his talk ["Philosophical Architecture in Grails"](https://www.youtube.com/watch?v=nx8XMY7evbA)
```
Comment (BirgitBader) [inline _posts/2016/2016-07-07-wiremock.md]: You can also directly implement the video like that:

{% youtube nx8XMY7evbA %}

File: _posts/2016/2016-07-07-wiremock.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+public class NoteServiceConsumerTest {
+
+  @Value("http://localhost:${wiremock.port}/") // (2)
+  String wiremockUrl;
+
+}
+```
+
+In the code example we annotate our consumer test with an annotation to start a WireMock server along with our test environment.
+By default WireMock gets started on a random port, and we need to point our API consumer to it, for example by using a Spring expression(2).
```
Comment (BirgitBader) [inline _posts/2016/2016-07-07-wiremock.md]: Spring expression (2)

The reference to (1) is missing in the text.

**Comment by BirgitBader** [comment]
Thanks for this awesome post, Olli! Review done.

File: _posts/2016/2016-07-07-wiremock.md (hunk 6 above)
Comment (otrosien) [inline _posts/2016/2016-07-07-wiremock.md]: Thought about it, but that would give too much weight/space on it, and I don't want the reader to stop and watch that video at that point. (I suspect he/she wouldn't finish reading then ;-)

File: _posts/2016/2016-07-07-wiremock.md (hunk 4 above)
Comment (otrosien) [inline _posts/2016/2016-07-07-wiremock.md]: ok, I'll need to create a new PR for that.

**Pull request closed without merging**